### PR TITLE
Refactor server.ts

### LINKED ISSUE #279: Server: Refactor server.ts
```Typescript
interface Server {
  init: (): express.Application => {};
  registerRoures: (app: express.Application) => {};
  isReady: (app: express.Application): boolean => {};
  start: (app: express.Application) => {};
}
```

### PR DESCRIPTION
**Pull request by sorsaffari**
## What is the goal of this PR?
To use a consistent structure as used for other routes and handlers for managing authentication endpoints, as well as, refactoring `server.ts` 

## What are the changes implemented in this PR?
closes #279 
- increases line max length to 160 `eslintrc.js`, which fits nicely on a 15-inch screen
-  adds `@types` for `express`, `cookie-session` and `express-graphql` as dev dependencies
- modifies `appRoot` and adds `userRoot` in `config.ts`
- moves `auth` routes to `src/routes/auth.ts`
- moves `auth` route handlers to `src/controllers/auth/index.ts`
- changes the `/pull_request` route to `/execution/pull-request` and move its handler  `src/controllers/auth/index.ts`
- implements `createInternal` in `ExecutionController`
- improves types on `getGraphqlServer` in `ExecutionController` and `SpanController`
- Refactors `server.ts`

### CODE REVIEW

File: service/web-server/src/server.ts [hunk 1]
```diff
@@ -1,203 +1,138 @@
...
-
-const LAUNCH_EXECUTOR_SCRIPT_PATH = `${__dirname}/../../launch_executor_server.sh`;
+if (server.isReady()) {
+    server.loadEnvVars();
+    server.registerRoutes();
+    server.start();
+} else {
+    Object.values(server.errors).forEach((error) => { console.log(error); });
+    process.exit(1);
+}
```
Comment (marco-scoppetta) [inline service/web-server/src/server.ts]: less defensive

File: service/web-server/src/server.ts [hunk 2]
```diff
@@ -1,203 +1,138 @@
...
-// setup and check for environment variables
-if (currentEnv === 'development') {
-    require('dotenv').config({ path: `${__dirname}/../.env` });
-} else if (currentEnv === 'production') {
-    require('dotenv').config({ path: '/home/ubuntu/.env' });
+function registerRoutes() {
+    // parse application/json
+    this.app.use(bodyParser.json());
+
+    this.esClient = getEsClient();
```
Comment (marco-scoppetta) [inline service/web-server/src/server.ts]: don't initialise twice and move the check in here (esClient.ping)

File: service/web-server/src/server.ts [hunk 3]
```diff
@@ -1,203 +1,138 @@
...
+
+    this.app.use('/auth', getAuthRoutes());
+
+    // Changes the requested location to the (default) /index.html, whenever there is a request which fulfills the following criteria:
+    // 1. The request is a GET request
+    // 2. which accepts text/html,
+    // 3. is not a direct file request, i.e. the requested path does not contain a . (DOT) character and
+    // 4. does not match a pattern provided in options.rewrites (read docs of connect-history-api-fallback)
+    this.app.use(history());
+    console.log(`${config.appRoot}../dashboard/dist`);
```
Comment (marco-scoppetta) [inline service/web-server/src/server.ts]: leftover

File: service/web-server/src/server.ts [hunk 4]
```diff
@@ -1,203 +1,138 @@
...
-// Is the ElasticSearch server running?
-esClient.ping((err) => {
-    if (err) {
-        console.error("Elastic Search Server is down. Makes sure it's up and running before starting the web-server.");
-        process.exit(1);
+function start() {
+    const invokedByScript = !module.parent;
+
+    if (invokedByScript) {
+        if (this.env === 'development') {
```
Comment (marco-scoppetta) [inline service/web-server/src/server.ts]: refactor

**Comment by sorsaffari** [comment]
@marco-scoppetta all resolved

**Pull request merged**